Reliant/Classes/ContextLocator/OCSDefaultContextRegistry.m

```diff
@@ -36,20 +36,62 @@ + (instancetype)sharedDefaultContextRegistry {
     return _instance;
 }
 
-- (void)registerContext:(id <OCSObjectContext>)context {
-    _contextRegister[context.name] = [OCSWeakWrapper weakWrapperWithObject:context];
+- (void)registerContext:(id <OCSObjectContext>)context toBoundObject:(NSObject *)boundObject {
+    NSString *contextName;
+
+    if(boundObject){
+        contextName = [NSString stringWithFormat:@"%@#%p", context.name, boundObject];
+    } else {
+        contextName = context.name;
+    }
+
+    _contextRegister[contextName] = [OCSWeakWrapper weakWrapperWithObject:context];
 }
 
-- (id <OCSObjectContext>)contextForName:(NSString *)name {
-    OCSWeakWrapper *weakRef = _contextRegister[name];
-    id<OCSObjectContext> result = nil;
-    if (weakRef) {
-        result = [weakRef object];
-        if (!result) {
-            [_contextRegister removeObjectForKey:name];
+- (id <OCSObjectContext>)contextForName:(NSString *)name fromBoundObject:(NSObject *)boundObject {
+    // Find all registry keys pertaining to the provided context name
+    NSArray *keys = [_contextRegister.allKeys filteredArrayUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(NSString * evaluatedObject, NSDictionary *bindings) {
+        return [evaluatedObject hasPrefix:name];
+    }]];
+
+    NSString *contextName;
+
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wundeclared-selector"
+
+    // More than 1 key candidate was found, attempt to deduce the most appropriate by inspecting the bound object's hierarchy
+    if(keys.count > 1 && boundObject && [boundObject respondsToSelector:@selector(parentViewController)]){
```

Comment:
nitpicking, but the second condition is obsolete, respondsToSelector will return false on nil